src/translator/response.h

```diff
@@ -40,21 +40,22 @@ struct Quality {
 /// AnnotatedText provides an API to access markings of (sub)-word and
 /// sentences boundaries, which are required to interpret Quality and
 /// Alignment (s) at the moment.
-class Response {
+struct Response {
```

Comment:
Few quick questions:
Does it help in some way to convert this class to a struct?
Do you want to keep all the data members of this class public?

Reply:
I have no strict notions here. Felt very natural to make this a struct and only carrying data/message to be responded with, and placing bindings on things like Annotation. 

If they're private, I'm probably looking at either setters or `ResponseBuilder` being a friend.

Reply:
I like having friends.

Reply:
Two things:

1, there's this API being used in a downstream app, where we will break API when members go private and accessors are needed.
2, This PR will become further bloated. Designing the proper API requires an iteration of experimentation ideally with WASM in my opinion, as several nuances like [this one](https://github.com/mozilla/bergamot-translator/issues/8#issuecomment-795929536) remain unanswered. I'm kind-of operating in the blind here without going into WASM which will take longer.

I'm not saying hard no-change here, just that it be deferred so certain other things can progress in parallel. Is this okay for everyone?

Reply:
Currently the primary user of this is mozilla, so changing the interface  should be consolidated with @abhi-agg and @motin .
What is the opinion from mozilla side?

Reply:
As this PR was intended to bridge the gap b/w Translation* classes and corresponding internal classes (Service, Response), we can go ahead without touching on Quality Scores and Alignments part of this structure.

> Designing the proper API requires an iteration of experimentation ideally with WASM in my opinion

I second this considering the Quality score and Alignment will require some experimentation on JS side 👍

My initial comment was about using `struct` vs a `class` and following a standard way of using the Builder design pattern in the code. I am fine to ignore both of these comments and go ahead 👍 

@jerinphilip Please let us know when the PR is ready for the review again so that we can expedite the merge 👍

Reply:
It's ready now. 

I have reverted a few additional DRY attempts to avoid bloating to keep diff minimum. I will be working on them separately after this PR (this will collapse the service-cli, service-cli-bytearray apps into one with a cmdline option and probably make emitting of the `Response` as JSON or something more granular to independently test quality / alignments / raw text / sentences outputs). The only changes now from last review is removal of the now empty `response.cpp` file and fixing service-cli-bytearray to account for `ResponseOptions`/`TranslationRequest`. Both of which are complete. 

Let me know if anything further needs to be done.